subsys/debug/Kconfig

```diff
@@ -43,6 +45,91 @@ config TRACING
 	  Enable system tracing. This requires a backend such as SEGGER
 	  Systemview to be enabled as well.
 
+if TRACING
+
+config TRACING_SYNC
+	bool "Synchronous Tracing"
+	help
+	  Enable synchronous tracing. This requires very little latency
+	  backend to be enabled as well.
+
+config TRACING_ASYNC
+	bool "Asynchronous Tracing"
+	default y
+	select RING_BUFFER
+	help
+	  Enable asynchronous tracing. This will buffer all the tracing
+	  packets to the ring buffer first, tracing thread will try to
+	  put as much data in ring buffer as possible when tracing thread
+	  get scheduled.
```

Comment:
what is the policy in case of overflow in the ring buffer? might be nice to write as part of help text.

Reply:
@mped-oticon currently for overflow case, just increase the dropped number. Basically I will add backend api(for usb backend, add one more endpoint; for uart backend I still considering it, maybe one more uart or console needed) specially for overflow message output so that it doesn't contaminate the normal stream data, I'm considering it since code developing, but not find a good way, I will enhance it after this PR merged.